include/deal.II/grid/geometric_entity_info.h

```diff
@@ -0,0 +1,359 @@
+    inline std_cxx20::ranges::iota_view<unsigned int, unsigned int>
+    line_indices() const
+    {
+      return {0U, n_lines()};
+    }
+
+    /**
+     * Return an object that can be thought of as an array containing all
+     * indices from zero to n_faces().
+     */
+    inline std_cxx20::ranges::iota_view<unsigned int, unsigned int>
+    face_indices() const
+    {
+      return {0U, n_faces()};
+    }
+
+    /**
+     * Standard decomposition of vertex index into face and face-vertex index.
+     */
+    virtual std::array<unsigned int, 2>
+    standard_vertex_to_face_and_vertex_index(const unsigned int vertex) const
+    {
+      Assert(false, ExcNotImplemented());
+
+      (void)vertex;
+
+      return {0u, 0u};
+    }
+
+    /**
+     * Standard decomposition of line index into face and face-line index.
+     */
+    virtual std::array<unsigned int, 2>
+    standard_line_to_face_and_line_index(const unsigned int line) const
+    {
+      Assert(false, ExcNotImplemented());
+
+      (void)line;
+
+      return {0, 0};
+    }
+
+    /**
+     * Correct vertex index depending on face orientation.
+     */
+    virtual unsigned int
+    standard_to_real_face_vertex(const unsigned int  vertex,
+                                 const unsigned int  face,
+                                 const unsigned char face_orientation) const
+    {
+      Assert(false, ExcNotImplemented());
+
+      (void)vertex;
+      (void)face;
+      (void)face_orientation;
+
+      return 0;
+    }
+
+    /**
+     * Correct line index depending on face orientation.
+     */
+    virtual unsigned int
+    standard_to_real_face_line(const unsigned int  line,
+                               const unsigned int  face,
+                               const unsigned char face_orientation) const
+    {
+      Assert(false, ExcNotImplemented());
+
+      (void)line;
+      (void)face;
+      (void)face_orientation;
+
+      return 0;
+    }
+
+    /**
+     * Combine face and line orientation.
+     */
+    virtual bool
+    combine_face_and_line_orientation(
+      const unsigned int  line,
+      const unsigned char face_orientation,
+      const unsigned char line_orientation) const
+    {
+      Assert(false, ExcNotImplemented());
+
+      (void)line;
+      (void)face_orientation;
+      (void)line_orientation;
+
+      return true;
+    }
+  };
+
+
+  /**
+   * Base class for tensor-product geometric entities.
+   */
+  template <int dim>
+  struct GeometricEntityInfoTensorProduct : GeometricEntityInfo
+  {
+    unsigned int
+    n_vertices() const override
+    {
+      return GeometryInfo<dim>::vertices_per_cell;
+    }
+
+    unsigned int
+    n_lines() const override
+    {
+      return GeometryInfo<dim>::lines_per_cell;
+    }
+
+    unsigned int
+    n_faces() const override
+    {
+      return GeometryInfo<dim>::faces_per_cell;
+    }
+  };
+
+
+
+  /*
+   * VERTEX
+   */
+  struct GeometricEntityInfoVertex : public GeometricEntityInfoTensorProduct<0>
```

Comment:
instead of long names, how about putting these all into a namespace `GeometricEntities` (or corresponding to whatever the class name is going to be)?

Reply:
I would be fine with a namespace `ReferenceCellInfo`. How would you call `GeometricEntityType` in PR #10637?

Reply:
I have introduced the namespace!